tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -67,23 +67,23 @@ function(mbed_set_toolchain_options target)
 endfunction()
 
 # GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
-# this step - linker options and compile definitions 
-function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+# this step - compile definitions
+function(mbed_generate_gcc_options_for_linker target definitions_file)
```

Comment:
This header does not describe what the function is doing. It should be something like:
```
# Generate a file containing compile definitions
```

How it does it (using generator expressions) is an implementation detail that should go inside the function. `GCC ARM requires preprecessing linker script` should be written where the function gets called in `mbed_set_mbed_target_linker_script`.

Reply:
fixed